### PR TITLE
Added oeedger8r tests for unsigned char, unsigned short, unsigned int & unsigned long

### PR DESCRIPTION
**Pull request by anitagov**
This is for Issue #978

### CODE REVIEW

**Comment by anitagov** [comment]
bors try

**Comment by anitagov** [comment]
bors try

**Comment by anitagov** [comment]
@anakrish - Incorporated your feedback. Please review.

File: tests/oeedger8r/edl/array.edl [hunk 1]
```diff
@@ -132,20 +132,41 @@ enclave  {
...
+        );
+
+        public void ecall_array_unsigned_int(
+            [in] unsigned int a1[2],
+            [in,out] unsigned int a2[2][2],
+            [out] unsigned int a3[3][3],
+            [user_check] unsigned int a4[4][4]
+        );
+
+	    public void ecall_array_unsigned_long(
```
Comment (anakrish) [inline tests/oeedger8r/edl/array.edl]: Nit: Indentation.

File: tests/oeedger8r/edl/array.edl [hunk 2]
```diff
@@ -278,8 +299,29 @@ enclave  {
...
+        );
+
+        void ocall_array_unsigned_int(
+            [in] unsigned int a1[2],
+            [in,out] unsigned int a2[2][2],
+            [out] unsigned int a3[3][3],
+            [user_check] unsigned int a4[4][4]
+        );
+
+	    void ocall_array_unsigned_long(
```
Comment (anakrish) [inline tests/oeedger8r/edl/array.edl]: Nit: Indentation

File: tests/oeedger8r/edl/pointer.edl [hunk 3]
```diff
@@ -598,7 +598,106 @@ enclave {
...
+            // size as parameter
+            [in, size=psize] unsigned int* p14,
+            [in,out, size=psize] unsigned int* p15,
+            [out, size=psize] unsigned int* p16,
+
+            int pcount,
+            int psize             
+        );
+
+	    public unsigned long* ecall_pointer_unsigned_long(
```
Comment (anakrish) [inline tests/oeedger8r/edl/pointer.edl]: Nit: Indentation

File: tests/oeedger8r/edl/pointer.edl [hunk 4]
```diff
@@ -1348,7 +1463,106 @@ enclave {
...
+            // size as parameter
+            [in, size=psize] unsigned int* p14,
+            [in,out, size=psize] unsigned int* p15,
+            [out, size=psize] unsigned int* p16,
+
+            int pcount,
+            int psize   
+        );
+
+	    unsigned long* ocall_pointer_unsigned_long(
```
Comment (anakrish) [inline tests/oeedger8r/edl/pointer.edl]: Nit indentation

File: tests/oeedger8r/enc/testarray.cpp [hunk 5]
```diff
@@ -302,11 +296,47 @@ void ecall_array_long_double(
...
+    unsigned long a2[2][2],
+    unsigned long a3[3][3],
+    unsigned long a4[4][4])
+{
+    ecall_array_fun_impl(a1, a2, a3, a4);
+}
+
 void ecall_array_assert_all_called()
 {
     // Each of the 16 functions above is called twice.
```
Comment (anakrish) [inline tests/oeedger8r/enc/testarray.cpp]: Change comment to 19

File: tests/oeedger8r/enc/testpointer.cpp [hunk 6]
```diff
@@ -1212,7 +1297,7 @@ void ecall_pointer_assert_all_called()
 {
     // Each of the 16 functions above is called twice.
```
Comment (anakrish) [inline tests/oeedger8r/enc/testpointer.cpp]: Change comment to 19

File: tests/oeedger8r/host/testpointer.cpp [hunk 7]
```diff
@@ -1206,7 +1333,7 @@ void ocall_pointer_assert_all_called()
 {
     // Each of the 16 functions above is called twice.
```
Comment (anakrish) [inline tests/oeedger8r/host/testpointer.cpp]: Change comment to 19

File: tests/oeedger8r/edl/array.edl (hunk 1 above)
Comment (anitagov) [reply tests/oeedger8r/edl/array.edl]: It looks fine in vi and Visual Studio Code. I used vi and changed the leading space (or tab) to tabs in vi and git reports it aligned now.

File: tests/oeedger8r/enc/testarray.cpp (hunk 5 above)
Comment (anitagov) [reply tests/oeedger8r/enc/testarray.cpp]: Fixed

File: tests/oeedger8r/edl/pointer.edl (hunk 4 above)
Comment (anitagov) [reply tests/oeedger8r/edl/pointer.edl]: Fixed

File: tests/oeedger8r/enc/testbasic.cpp [hunk 8]
```diff
@@ -178,7 +206,11 @@ void ecall_basic_types(
     uint16_t arg13,
     uint32_t arg14,
     uint64_t arg15,
-    long long arg16)
+    long long arg16,
+    unsigned char arg17,
+    unsigned short arg18,
+    unsigned int arg19,
+    unsigned long arg20)
```
Comment (anitagov) [inline tests/oeedger8r/enc/testbasic.cpp]: arg20 has been deleted.

File: tests/oeedger8r/enc/testbasic.cpp [hunk 9]
```diff
@@ -199,6 +231,10 @@ void ecall_basic_types(
     check_type<uint32_t>(args.arg14);
     check_type<uint64_t>(args.arg15);
     check_type<long long>(args.arg16);
+    check_type<unsigned char>(args.arg17);
+    check_type<unsigned short>(args.arg18);
+    check_type<unsigned int>(args.arg19);
+    check_type<unsigned long>(args.arg20);
```
Comment (anitagov) [inline tests/oeedger8r/enc/testbasic.cpp]: Delete this line

File: tests/oeedger8r/enc/testbasic.cpp [hunk 10]
```diff
@@ -216,6 +252,10 @@ void ecall_basic_types(
     OE_TEST(arg14 = 16);
     OE_TEST(arg15 = 17);
     OE_TEST(arg16 = 18);
+    OE_TEST(arg17 = 255);
+    OE_TEST(arg18 = 19);
+    OE_TEST(arg19 = 20);
+    OE_TEST(arg20 = 21);
```
Comment (anitagov) [inline tests/oeedger8r/enc/testbasic.cpp]: This line needs to be deleted.

File: tests/oeedger8r/host/testbasic.cpp [hunk 11]
```diff
@@ -126,6 +130,30 @@ void test_basic_edl_ecalls(oe_enclave_t* enclave)
...
+        OE_TEST(ret == 191);
+    }
+
+    {
+        unsigned int ret = 0;
+        OE_TEST(ecall_ret_unsigned_int(enclave, &ret) == OE_OK);
+        OE_TEST(ret == 202);
+    }
+
+    {
```
Comment (anitagov) [inline tests/oeedger8r/host/testbasic.cpp]: Add if (g_enabled[TYPE_UNSIGNED_LONG]) before this.

File: tests/oeedger8r/host/testbasic.cpp [hunk 12]
```diff
@@ -182,7 +210,11 @@ void ocall_basic_types(
     uint16_t arg13,
     uint32_t arg14,
     uint64_t arg15,
-    long long arg16)
+    long long arg16,
+    unsigned char arg17,
+    unsigned short arg18,
+    unsigned int arg19,
+    unsigned long arg20)
```
Comment (anitagov) [inline tests/oeedger8r/host/testbasic.cpp]: Delete this line.

File: tests/oeedger8r/host/testbasic.cpp [hunk 13]
```diff
@@ -203,6 +235,10 @@ void ocall_basic_types(
     check_type<uint32_t>(args.arg14);
     check_type<uint64_t>(args.arg15);
     check_type<long long>(args.arg16);
+    check_type<unsigned char>(args.arg17);
+    check_type<unsigned short>(args.arg18);
+    check_type<unsigned int>(args.arg19);
+    check_type<unsigned long>(args.arg20);
```
Comment (anitagov) [inline tests/oeedger8r/host/testbasic.cpp]: This line is gone now too.

File: tests/oeedger8r/host/testbasic.cpp [hunk 14]
```diff
@@ -220,6 +256,10 @@ void ocall_basic_types(
     OE_TEST(arg14 = 16);
     OE_TEST(arg15 = 17);
     OE_TEST(arg16 = 18);
+    OE_TEST(arg17 = 255);
+    OE_TEST(arg18 = 19);
+    OE_TEST(arg19 = 20);
+    OE_TEST(arg20 = 21);
```
Comment (anitagov) [inline tests/oeedger8r/host/testbasic.cpp]: You can delete arg20

**Pull request merged**